string_ascii.go

```diff
@@ -243,6 +243,11 @@ func (s asciiString) hash(hash *maphash.Hash) uint64 {
 }
 
 func (s asciiString) charAt(idx int) rune {
+	for i, r := range s {
+		if i == idx {
+			return r
```

Comment:
A small explanation, this will return a rune, the original return would convert a `byte` to a rune. Why would that not work ?

My guess is that while the for loop catches every rune one by one the index on a string accesses the byte which for some special characters does not necessarily correspond to a rune (in this case the first byte was "half" of a rune).

Reply:
Just a passer-by comment: this is the asciiString, it should not have any non-ascii symbols in it, so the previous code should've been fine.

I would guess the bug comes from somewhere where a Unicode character gets added to the asciiString or an improper initialization of one. I don't know if this is due to the changes in this branch or if it's an original goja issue, but if you think it's the later it might be better to report it.

Also I will be interested to help you if you want to make PRs for at least some of the changes in this branch.

Reply:
Comment much appreciated! If you look at the snippet this only happens with data returned by the `findOne` request so I think that might be returning what you mentioned a Unicode character added to the asciiString. I was trying to have a test to specifically reproduce this without the call to `findOne` and might get around to do that this week.

I'm not sure if this bug is on goja or in how we return data with `findOne`, happy to report this as soon as I have a way to reproduce it without `findOne` and eventually submit a PR. Some other changes in this branch are "custom" which would only pollute `goja` with useless stuff for the general public =D